x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_management/api/rules/bulk_actions/fetch_rules_by_query_or_ids.ts

```diff
@@ -7,39 +7,39 @@
 
 import type { RulesClient } from '@kbn/alerting-plugin/server';
 import { BadRequestError } from '@kbn/securitysolution-es-utils';
-import { MAX_RULES_TO_UPDATE_IN_PARALLEL } from '../../../../../../../common/constants';
-import type { PromisePoolOutcome } from '../../../../../../utils/promise_pool';
-import { initPromisePool } from '../../../../../../utils/promise_pool';
+import { type PromisePoolOutcome } from '../../../../../../utils/promise_pool';
 import type { RuleAlertType } from '../../../../rule_schema';
-import { readRules } from '../../../logic/detection_rules_client/read_rules';
 import { findRules } from '../../../logic/search/find_rules';
 
 export const fetchRulesByQueryOrIds = async ({
   query,
   ids,
   rulesClient,
-  abortSignal,
   maxRules,
 }: {
   query: string | undefined;
   ids: string[] | undefined;
   rulesClient: RulesClient;
-  abortSignal: AbortSignal;
   maxRules: number;
 }): Promise<PromisePoolOutcome<string, RuleAlertType>> => {
   if (ids) {
-    return initPromisePool({
-      concurrency: MAX_RULES_TO_UPDATE_IN_PARALLEL,
-      items: ids,
-      executor: async (id: string) => {
-        const rule = await readRules({ id, rulesClient, ruleId: undefined });
-        if (rule == null) {
-          throw Error('Rule not found');
+    const { rules, errors } = await rulesClient.bulkGetRules({ ids });
+    return {
+      results: rules.map((rule) => ({
+        item: rule.id,
+        result: rule,
+      })),
+      errors: errors.map(({ id, error }) => {
+        let message = 'Error resolving the rule';
+        if (error.statusCode === 404) {
+          message = 'Rule not found';
         }
-        return rule;
-      },
-      abortSignal,
```

Comment:
@xcrzx I seem to remember you doing work on bulk action requests timing out, do you think removing the `abortSignal` here is problematic?

Reply:
Yeah, I remember we had issues with the route timing out in the past but I guess using the rulesClient path directly should subvert those issues 👍